### ISSUE TITLE
Segfault when viewing pixel history

### ISSUE THREAD

**Issue by kvark** [baldurk/renderdoc#2280]
## Description

Checking the pixel history crashes RD.

## Steps to reproduce

1. Open [jasper.rdc.zip](https://github.com/baldurk/renderdoc/files/6544071/jasper.rdc.zip)
2. navigate to draw call 330
3. click "History" for any selected pixel

## Environment

* RenderDoc version: 1.13
* Operating System: Linux (Solus distro)
* Graphics API: Vulkan

GPU: Intel Corporation UHD Graphics 630 (Mobile)

**Comment by baldurk** [comment]
I can't reproduce the crash, but the capture you provided was made on an nvidia GPU not an Intel one so I'm not sure if you uploaded the wrong capture? Since the filename is `intel.rdc` and you said that you were using an Intel GPU in your report.

I only tested on windows/nvidia because my linux/nvidia system is heavily out of date. I'm updating it now in the background but I don't think there's a high chance that this will reproduce there given nvidia tends to use largely the same driver on both OSs.

Can you say exactly which nvidia driver you're using? The capture was made on 465.27, is that what you're using - can you try updating to a more recent driver?

**Comment by kvark** [comment]
Oh, sorry about the confusion! Now trying to force Intel in RD's options, it fails to launch the app - 
[rd-intel.log](https://github.com/baldurk/renderdoc/files/6561699/rd-intel.log)

We've also got people (like @cwfitzgerald) being able to load the RD capture and see the pixel history, but it shows nonsensical stuff like
![pixel history](https://matrix-client.matrix.org/_matrix/media/r0/download/matrix.org/ZoeTUNUgRKchwpgshHHCGFiT)

(it says "depth clipped", but also that the texture value changes, and the depth value shouldn't be clipped there)

Here are the launch instructions for the app, in case you want to see it yourself.
```
# install git, rust, and cargo
git clone https://github.com/gfx-rs/wgpu
cd wgpu/player
cargo run -- <trace_folder>
```
Where the "trace_folder" can be extracted from 
[wgpu-trace.zip](https://github.com/baldurk/renderdoc/files/6561712/wgpu-trace.zip)

This will build the app and run it, also spewing the command line you could put into RD:
> Running `/hub/moz/wgpu/target/debug/play /hub/bugs/jasper-bug/wgpu-trace/`

**Comment by baldurk** [comment]
I don't understand what you mean about forcing Intel. RenderDoc has no way of forcing an application to use a particular GPU, and the log you posted doesn't show any indication that a particular GPU is being forced for replay. The nvidia GPU is getting picked up as the closest match.

Jasper has reported the issue to me separately. Something in the capture is non-deterministic which breaks the pixel history as it relies on repeatable results. From what Jasper has said this happens in the application as well so it doesn't seem like a RenderDoc bug.

Please don't move the goalposts and try to change what the issue is about. Are you actually seeing a crash when running the pixel history and can you explain exactly how you get it and how to reproduce? You said that it happened with the original capture but I can't reproduce that - are you able to get a crash with the original capture yourself when running on an nvidia GPU? If so what driver are you running on? The log you posted shows a device lost error but not a crash so I'm unclear on if that's the same bug or a different one.

**Comment by kvark** [comment]
I'm talking about "Settings ->Replay -> GPU Selection Override" thing.

> Please don't move the goalposts and try to change what the issue is about.

I don't see any goalposts moved. Clicking on pixel history crashes RD for me, it's still the original issue. The extra info I provided about nonsensical results are to help investigating this. My assumption was that RD goes into some kind of undefined behavior, which results in a crash for me but garbage on other machines.

> You said that it happened with the original capture but I can't reproduce that - are you able to get a crash with the original capture yourself when running on an nvidia GPU? If so what driver are you running on? The log you posted shows a device lost error but not a crash so I'm unclear on if that's the same bug or a different one.

Yes, it crashes when doing the steps described in the body of this issue. My NV driver version in 465.27.

**Comment by baldurk** [comment]
As indicated by the section of the settings, that override only applies to the replay and has no effect on what physical device the application selects.

The inconsistent contents of the pixel history are not related to the crash and they match up with the capture itself and indeed the application which behaves unpredictably or produces nonsense results. That's why I didn't want to confuse the issue and stick just to the crash. From talking to Jasper separately that problem looks like an nv shader compiler bug.

I was able to get my nvidia linux system updated and did repro the crash. I'm actually not sure how I was able to open it on windows (or the other person you mentioned), but this looks like the bug fixed by #2244 with pixel history on imageless framebuffers. If you update to a recent nightly build or preferably to the v1.14 released since this bug was opened that should work correctly.
